tests/backup/backup_test.go

```diff
@@ -347,21 +349,8 @@ var _ = Describe("{DuplicateSharedBackup}", func() {
 		backupDeleteResponse, err := DeleteBackup(backupName, backupUID, orgID, ctx)
 		log.FailOnError(err, "Backup [%s] could not be deleted with delete response %s", backupName, backupDeleteResponse)
 
-		log.Infof("Deleting registered clusters for admin context")
-		err = DeleteCluster(SourceClusterName, orgID, ctx)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
-		err = DeleteCluster(destinationClusterName, orgID, ctx)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", destinationClusterName))
-
-		log.Infof("Cleaning up backup location - %s", backupLocationName)
-		err = DeleteBackupLocation(backupLocationName, backupLocationUID, orgID)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", backupLocationName))
-
-		log.Infof("Cleaning cloud credential")
-		time.Sleep(time.Minute * 3)
-		err = DeleteCloudCredential(credName, orgID, cloudCredUID)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cloud cred %s", credName))
-
+		log.Infof("Cleaning cloud accounts")
+		DeleteCloudAccounts(backupLocationMap, credName, cloudCredUID, ctx)
```

Comment:
In my opinion, it would be appropriate to include error handling in the DeleteCloudAccounts function to check if backupLocationMap is empty and return an error if necessary, similar to the error handling in the DeleteCloudCredential function. This can help ensure the proper functioning.

Reply:
Normally in an helper function we are not handling any error. We are just returning it to the caller and the caller has to handle it. But here `DeleteCloudAccounts` is a cleanup function and there is nothing to handle outside for the caller. If there is an error we are doing a VerifySafely inside the function itself. This is an exception only for cleanup functions. Hence we are not returning error.

Reply:
Makes sense.